backend/src/main/resources/db/changelog/db.changelog-master.yaml

```diff
@@ -1209,4 +1212,233 @@ databaseChangeLog:
+                    referencedTableName: device_specimen
+        - addColumn:
+            tableName: test_order
+            columns:
+              - column:
+                  name: device_specimen_id
+                  remarks: The device/specimen combination used to record the test result.
+                  type: *idtype
+                  constraints:
+                    foreignKeyName: fk__test_order__device_specimen
+                    referencedTableName: device_specimen
+                    # will be made non-null in a subsequent migration
+        - addColumn:
+            tableName: test_event
+            columns:
+              - column:
+                  name: device_specimen_id
+                  remarks: The device/specimen combination used to record the test result.
+                  type: *idtype
+                  constraints:
+                    foreignKeyName: fk__test_event__device_specimen
+                    referencedTableName: device_specimen
+                    # will be made non-null in a subsequent migration
+  - changeSet:
+      id: create-migrations-user
+      author: bwarfield@cdc.gov
+      comment: Create a system-level user to attribute migration insert/updates to.
+      changes:
+        - sql:
+            remarks: Create migrations user to indicate records created by migration rather than by a human.
+            sql: |
+              INSERT INTO ${database.defaultSchemaName}.api_user
+              (internal_id, created_at, updated_at, last_seen, login_email, last_name)
+              VALUES (
+                gen_random_uuid(), now(), now(), null,
+                '${migrations_user_email}', 'Database Migration Process'
+              )
+  - changeSet:
+      id: populate-specimen-tables
+      author: bwarfield@cdc.gov
+      comment: Populate specimen tables and relations based on existing data.
+      preConditions:
+        - onSqlOutput: FAIL
+        - onFail: MARK_RAN
+        - sqlCheck:
+            sql:  SELECT case when count(1) > 0 then 1 else 0 end FROM ${database.defaultSchemaName}.device_type;
+            expectedResult: 1
+      changes:
+        - sql:
+            remarks: Create in-use specimen types based on current device-type data.
+            sql: |
+              WITH swab_types_historical as (
+                select distinct swab_type as type_code from ${database.defaultSchemaName}.device_type
+              ),
+              migration_user as (
+                SELECT internal_id as migrations_user_id
+                FROM ${database.defaultSchemaName}.api_user
+                WHERE login_email = '${migrations_user_email}'
+              )
+              INSERT INTO ${database.defaultSchemaName}.specimen_type (name, type_code,
+                internal_id, created_by, created_at, updated_by, updated_at, is_deleted)
+              SELECT
+                'Auto-generated specimen type ' || type_code,
```

Comment:
i see multiple uses of this "Auto-generated " language. Will users see this? This doesn't feel like language users should see.

Reply:
Not in the short term—if, by the time we are rolling out the changes that expose that value in the UI, nobody has taken the time to update the placeholder values to the correct ones, then yes customers could see it, but that's pretty easy to prevent.